### ISSUE TITLE
Provide votes/finalVotes information in GET proposals/proposalId response

### ISSUE THREAD

**Issue by gaurav137** [microsoft/CCF#6107]
For building a user experience around proposals we needed the following ability:

1.  For open proposals show the yes/no vote status for each member that has voted till now.
2. For closed proposals show the yes/no vote status for how each member had voted
3. For open proposals show which members are yet to vote.

Based on a discussion with @eddyashton it looks like that 1. and 2. might require some changes from CCF's end to expose the "votes" or "finalVotes" property in the GET proposals/proposalId response. And 3. is something we'd have to deduce by say enumerating all current members and picking members that are not present in the votes/finalVotes object.

**Comment by eddyashton** [comment]
Update after further discussion:

1. It's tricky to show _votes_ for **open** proposals, since the _votes_ (boolean result) might change at any time. But we can show the IDs of the members who have voted (and separately, we retain the separate endpoint that returns the maybe-large
2. We can and should do this - looks like a bug that it's occasionally omitted, which we'll fix.
3. This is out-of-scope at the framework level, as the set of appropriate members depends on the resolve logic. Simpler for the client logic to construct this from its list of all (appropriate) members, and a `hasVoted` list we should include in part 1.

**Comment by achamayou** [comment]
Update following discussion with @gaurav137:

1. We want to add a field to `GET /gov/members/proposals/{proposalId}` listed the memberIds that have submitted a ballot for the proposal so far.
2. We want to confirm that `finalVotes` is always being populated correctly. Edit: that looks like it's already covered by [governance_js.py:test_final_proposal_visibility()](https://github.com/microsoft/CCF/blob/d5442e75dc5037de58c0781e17959d379b29ba8b/tests/governance_js.py#L1311), done in #6114.

**Comment by achamayou** [comment]
Naming proposal from GH Copilot:

> The existing fields in the proposal response are:
> 
> | Field | Type | Description |
> |---|---|---|
> | `proposalId` | string | ID of the proposal |
> | `proposerId` | string | member ID of the proposer |
> | `proposalState` | string | current state |
> | `ballotCount` | integer | number of ballots submitted |
> | `finalVotes` | object | member ID → bool (only after resolution) |
> | `voteFailures` | object | member ID → failure (only after resolution) |
> | `failure` | object | failure info (only on failure) |
> 
> The naming convention is **camelCase**, noun-based, and ballot-related fields use the term "ballot" (see `ballotCount`). The field you want is the list of member IDs that have submitted a ballot — essentially the keys of the `ballots` map while the proposal is still open.
> 
> The best name would be **`ballotSubmitters`**.
> 
> Reasoning:
> - It uses the established "ballot" terminology (matching `ballotCount`)
> - It describes *who* submitted, not a count or result
> - It's parallel to how `finalVotes` / `voteFailures` name their concepts — a noun describing the collection
> - Alternatives like `ballotMemberIds` are more verbose and inconsistent with the style (no other field appends `Id`/`Ids` to describe a collection)
> - `voterIds` would clash with the separate "vote" vs "ballot" distinction the API already maintains (`ballotCount` not `voteCount`)

**Comment by eddyashton** [comment]
Should be resolved by #7840, released in https://github.com/microsoft/CCF/releases/tag/ccf-7.0.2. As discussed above, we want to deal with _ballots_ (the input script) rather than the evaluated _vote_ (the output result), since that's what is stable and currently stored in the KV. We now include `ballotSubmitters`, allowing those ballots to be looked up from the existing [`GET /gov/members/proposals/{proposalId}/ballots/{memberId}`](https://ccf.dev/main/governance/gov_api_schemas/2024-07-01.html#get--gov-members-proposals-proposalId-ballots-memberId] endpoint. Where those ballots are trivial `return true`/`return false` scripts, that's equivalent to a `getVote`. In other cases, it's up to the client to interpret.
